Develop convert to pygi

doadin opened the pull request.

bendikro committed:
Fixed most PYGI issues in path_combo_chooser.py

Conflicts:
	deluge/ui/gtkui/path_combo_chooser.py

bendikro committed:
HACK

Conflicts:
	deluge/ui/gtkui/path_combo_chooser.py

doadin committed:
[#2743][GTKUI] Convert  to pygi/gtk3

doadin committed:
Address some flake8 and some left over TOFIX/TODO

doadin committed:
f to i

doadin committed:
i to f

doadin committed:
snow flakes

doadin committed:
yellow snow

doadin committed:
test snow

doadin committed:
test snow 1

doadin committed:
f

doadin committed:
flakes

doadin committed:
travis updates

doadin committed:
flake test1

doadin committed:
flake test2

doadin committed:
flake 3

doadin committed:
flake 4

doadin committed:
flake5

doadin committed:
Fix dbus

At the very least stop running it on windows as it doesnt close the dbus
process.

doadin committed:
Fix a TOFIX

Code is changed elsewhere and is no longer needed. Slight change but
same exact result.

doadin committed:
Fix formating in main window

Set line wraping so it visually looks better

doadin committed:
Fix error due to missing import

doadin committed:
Revert "Fix error due to missing import"

This reverts commit a1ed56be4e1febfc4ed87bbd0790e82ee2d58e40.

doadin committed:
because

doadin committed:
p 1

doadin committed:
G 1

doadin commented on deluge/ui/gtkui/filtertreeview.py:
```diff
@@ -99,12 +99,12 @@ class "GtkTreeView" style "treeview-style" """)
-        self.colour_background = style.bg[gtk.STATE_NORMAL]
-        self.colour_foreground = style.fg[gtk.STATE_NORMAL]
+        style = self.window.window.get_style().copy()
+        self.colour_background = self.window.window.modify_bg(Gtk.StateType.NORMAL)
```
?

doadin committed:
cleanup

doadin committed:
Cleanup 2

doadin committed:
test?

doadin committed:
test2?

doadin committed:
Revert "test2?"

This reverts commit b950d54bc22e1651efc9d45627e9cf7af78289b5.

doadin committed:
Revert "test?"

This reverts commit 7f4cfdeea7974e6d6ad1df2008a4747df4308d40.

doadin committed:
Fix autoadd ui

doadin committed:
Fix depricated item in autoadd ui

doadin committed:
Fix Another depricated ui item and cleanup old glade files

doadin committed:
Remove depricated items from main ui parts

doadin commented on deluge/plugins/AutoAdd/deluge/plugins/autoadd/gtkui.py:
```diff
@@ -40,132 +39,133 @@ class OptionsDialog():
         label_widget.set_model(self.labels)
-        label_widget.set_text_column(0)
-        self.glade.get_widget('label_toggle').set_active(options.get('label_toggle', False))
+        # label_widget.set_text_column(0) TOFIX
```
AttributeError:  'ComboBox' object has no attribute 'set_text_column'

doadin commented on deluge/plugins/AutoAdd/deluge/plugins/autoadd/gtkui.py:
```diff
@@ -40,132 +39,133 @@ class OptionsDialog():
-        label_widget = self.glade.get_widget('label')
-        label_widget.child.set_text(options.get('label', ''))
+        label_widget = self.main_builder.get_object('label')
+        # label_widget.get_child().set_text(options.get('label', '')) TOFIX
```
AttributeError: 'CellView' object has no attribute 'set_text'

doadin commented on deluge/plugins/AutoAdd/deluge/plugins/autoadd/gtkui.py:
```diff
@@ -299,43 +299,43 @@ def on_cancel(self, event=None):
 
-        options['label'] = self.glade.get_widget('label').child.get_text().lower()
-        options['append_extension'] = self.glade.get_widget('append_extension').get_text()
+        # options['label'] = self.main_builder.get_object('label').get_child().get_text().lower() TOFIX
```
AttributeError: 'CellView' object has no attribute 'get_text'

doadin commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -176,20 +177,20 @@ def load_preferences_page(self):
+        renderer = Gtk.CellRendererText()
         renderer.connect("edited", self.on_cell_edited, self.whitelist_model)
-        renderer.set_data("ip", 0)
+        # renderer.set_data("ip", 0) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -271,48 +270,41 @@ def enable(self):
         renderer.connect("edited", self.on_cell_edited, self.recipients_model)
-        renderer.set_data("recipient", RECIPIENT_FIELD)
-        column = gtk.TreeViewColumn("Recipients", renderer,
+        # renderer.set_data("recipient", RECIPIENT_FIELD) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -321,35 +313,35 @@ def build_recipients_model_populate_treeview(self):
-        renderer.set_data("event", SND_EVENT)
-        column = gtk.TreeViewColumn("Event", renderer, text=SND_EVENT)
+        renderer = Gtk.CellRendererText()
+        # renderer.set_data("event", SND_EVENT) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -321,35 +313,35 @@ def build_recipients_model_populate_treeview(self):
-        renderer.set_data("event_doc", SND_EVENT_DOC)
-        column = gtk.TreeViewColumn("Doc", renderer, text=SND_EVENT_DOC)
+        renderer = Gtk.CellRendererText()
+        # renderer.set_data("event_doc", SND_EVENT_DOC) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -321,35 +313,35 @@ def build_recipients_model_populate_treeview(self):
-        renderer.set_data("sound_name", SND_NAME)
-        column = gtk.TreeViewColumn("Name", renderer, text=SND_NAME)
+        renderer = Gtk.CellRendererText()
+        # renderer.set_data("sound_name", SND_NAME) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -321,35 +313,35 @@ def build_recipients_model_populate_treeview(self):
-        renderer.set_data("sound_path", SND_PATH)
-        column = gtk.TreeViewColumn("Path", renderer, text=SND_PATH)
+        renderer = Gtk.CellRendererText()
+        # renderer.set_data("sound_path", SND_PATH) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -357,51 +349,51 @@ def build_sounds_model_populate_treeview(self):
-        renderer.set_data("event", SUB_EVENT)
-        column = gtk.TreeViewColumn("Event", renderer, text=SUB_EVENT)
+        renderer = Gtk.CellRendererText()
+        # renderer.set_data("event", SUB_EVENT) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -357,51 +349,51 @@ def build_sounds_model_populate_treeview(self):
-        renderer.set_data("event_doc", SUB_EVENT)
-        column = gtk.TreeViewColumn("Doc", renderer, text=SUB_EVENT_DOC)
+        renderer = Gtk.CellRendererText()
+        # renderer.set_data("event_doc", SUB_EVENT) TOFIX
```
RuntimeError: Data access methods are unsupported. Use normal Python attributes instead

doadin committed:
Update gtkui.py

doadin committed:
Update gtkui.py

doadin committed:
Update listview.py

doadin committed:
Update gtkui.py

doadin committed:
Update common.py

doadin committed:
Update gtkui.py

doadin committed:
Update gtkui.py

doadin committed:
GObject related things

cas-- commented on deluge/plugins/AutoAdd/deluge/plugins/autoadd/gtkui.py:
```diff
@@ -299,43 +299,42 @@ def on_cancel(self, event=None):
+            options['copy_torrent'] = self.main_builder.get_object(
                 'copy_torrent_entry').get_text()
 
-        options['label'] = self.glade.get_widget('label').child.get_text().lower()
```
This does not look correct

doadin replied:
should be more like 
options['label'] = self.main_builder.get_object('label').child.get_text().lower() 
however you get
AttributeError: 'ComboBox' object has no attribute 'child'

deffinatly not correct and needs a TODO

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -53,58 +53,58 @@ def disable(self):
         # Remove status item
         component.get("StatusBar").remove_item(self.status_item)
-        del self.status_item
+        # del self.status_item
```
why?

doadin replied:
fixed

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -53,58 +53,58 @@ def disable(self):
         self.plugin.deregister_hook("on_show_prefs", self._on_show_prefs)
 
-        del self.glade
+        # del self.glade
```
why?

doadin replied:
fixed

cas-- replied:
Looking at the code again it should be `del self.main_builder`

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -141,20 +141,21 @@ def _on_status_item_clicked(self, widget, event):
-        self.progress_bar = self.glade.get_widget("progressbar")
-        self.table_info = self.glade.get_widget("table_info")
+        self.whitelist_frame = self.main_builder.get_object("whitelist_frame")
+        self.progress = Gtk.ProgressBar()
```
should be `self.progress_bar`

doadin replied:
fixed

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -141,20 +141,21 @@ def _on_status_item_clicked(self, widget, event):
 
         # Hide the progress bar initially
-        self.progress_bar.hide()
+        self.progress.hide()
```
should be `self.progress_bar`

doadin replied:
fixed

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -53,58 +53,58 @@ def disable(self):
-                self.progress_bar.set_text("Downloading %.2f%%" % (status["file_progress"] * 100))
-                self.progress_bar.set_fraction(status["file_progress"])
-                self.progress_bar.show()
+                self.progress.set_text("Downloading %.2f%%" % (status["file_progress"] * 100))
```
should be `self.progress_bar`

doadin replied:
fixed

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -53,58 +53,58 @@ def disable(self):
-                self.progress_bar.set_text("Importing %s" % (status["num_blocked"]))
-                self.progress_bar.pulse()
-                self.progress_bar.show()
+                self.progress.set_text("Importing %s" % (status["num_blocked"]))
```
should be `self.progress_bar`

doadin replied:
fixed

cas-- commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -159,15 +158,14 @@ def __blink(self):
+            self.note = Notify.Notification(title, message)
             self.note.set_icon_from_pixbuf(icon)
             if not self.note.show():
                 err_msg = _("pynotify failed to show notification")
```
This message needs updated

doadin replied:
updated

cas-- commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -159,15 +158,14 @@ def __blink(self):
-            self.note = pynotify.Notification(title, message)
+        if Notify.init("Deluge"):
+            icon = GdkPixbuf.Pixbuf.new_from_file_at_size(deluge.common.get_pixmap("deluge.svg"), 48, 48)
+            self.note = Notify.Notification(title, message)
```
This might need extra code, looking at similar changes in another project: https://github.com/eonpatapon/mpDris2/commit/c1a7e9741f419836b307d8d1514a68350da620ea#diff-04de6ed057889d6b920e21aa4aa92919R790

doadin replied:
needs todo i havnt really tested this plugin extensivly ill look into it

cas-- commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -231,11 +229,12 @@ def __init__(self, plugin_name):
     def enable(self):
         self.config = deluge.configmanager.ConfigManager(
-            "notifications-gtk.conf", DEFAULT_PREFS
+            "notifications-Gtk.conf", DEFAULT_PREFS
```
This is a wrong

doadin replied:
fixed

cas-- commented on deluge/ui/gtkui/glade/add_torrent_dialog.ui:
```diff
@@ -313,8 +361,6 @@
-                    <property name="vscrollbar_policy">automatic</property>
                     <property name="shadow_type">out</property>
                     <child>
                       <object class="GtkTreeView" id="listview_files">
```
To make it clearer what is being references in code, should rename `listview_files` to `treeview_files` and obviously change all references in  `deluge/ui/gtkui/addtorrentdialog.py`

doadin replied:
changed ...fixed

cas-- commented on deluge/ui/gtkui/glade/add_torrent_dialog.ui:
```diff
@@ -56,15 +102,14 @@
-                            <property name="vscrollbar_policy">automatic</property>
                             <property name="shadow_type">in</property>
                             <child>
                               <object class="GtkTreeView" id="listview_torrents">
```
To make it clearer what is being referenced in code, should rename `listview_torrents` to `treeview_torrents` and obviously change all references in  `deluge/ui/gtkui/addtorrentdialog.py`

cas-- commented on deluge/plugins/Scheduler/deluge/plugins/scheduler/gtkui.py:
```diff
@@ -13,26 +13,30 @@
 import deluge.component as component
 from deluge.plugins.pluginbase import GtkPluginBase
 from deluge.ui.client import client
+from gi import pygtkcompat
```
I think it is now `import pygtkcompat` (see https://twistedmatrix.com/trac/ticket/6280) but is it still needed in this plugin?

doadin replied:
fixed and yes theres still something(cant recall right now) that isnt updated for the api change fairly small change but still .. needs TODO

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -25,22 +23,30 @@
-                    'bandwidth_graph': {'upload_rate': str(gtk.gdk.Color("blue")),
-                                        'download_rate': str(gtk.gdk.Color("green")),
+                    'bandwidth_graph': {'upload_rate':
+                                        str(Gdk.RGBA(red=0.541176, green=0.886275, blue=0.203922, alpha=1.000000)),
```
I think should be using: https://developer.gnome.org/gdk3/stable/gdk3-RGBA-Colors.html#gdk-rgba-parse

doadin replied:
i think i might have tried that at some point in my testing? i can try again but do we have to?

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -66,22 +72,22 @@ def int_str(number):
-    red = gtk_color.red / max_val
-    green = gtk_color.green / max_val
-    blue = gtk_color.blue / max_val
+    """Turns a Gdk.Color into a tuple with range 0-1 as used by the graph"""
```
Would [gdk_rgba_to_string](https://developer.gnome.org/gdk3/stable/gdk3-RGBA-Colors.html#gdk-rgba-to-string) be a better replacement here?

doadin replied:
that it would....but apparently i dont know how... FIXME

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -107,21 +113,23 @@ def __init__(self, glade, colors):
-        context.rectangle(event.area.x, event.area.y, event.area.width, event.area.height)
+        alloc = Gtk.DrawingArea.get_allocation(widget)
+        w, h = alloc.width, alloc.height
+        context.rectangle(0, 0, w, h)
```
Is `0,0` correct replacement for `event.area.x, event.area.y`

doadin replied:
i beleive so. if i understand correctly and remeber correctly i bleieve this just starts the graph? in the top left corner.

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -107,21 +113,23 @@ def __init__(self, glade, colors):
+        context = self.graph_widget.get_property('window').cairo_create()
         # set a clip region
-        context.rectangle(event.area.x, event.area.y, event.area.width, event.area.height)
+        alloc = Gtk.DrawingArea.get_allocation(widget)
```
Are you sure this code works the same as `event.area`?

doadin replied:
yes with 99% certainty?

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -107,21 +113,23 @@ def __init__(self, glade, colors):
         self.graph.draw_to_context(context,
-                                   self.graph_widget.allocation.width,
-                                   self.graph_widget.allocation.height)
+                                   w,
```
No need to be on separate lines if this is the correct code replacement.

doadin replied:
fixed

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -216,13 +224,15 @@ class GtkUI(GtkPluginBase):
-        self.glade = XML(common.get_resource("config.glade"))
-        component.get("Preferences").add_page("Stats", self.glade.get_widget("prefs_box"))
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("config.ui"))
```
`self.glade =` not neded here, just `self.main_builder...`

doadin replied:
fixed

cas-- commented on deluge/plugins/Stats/deluge/plugins/stats/gtkui.py:
```diff
@@ -253,8 +263,9 @@ def on_show_prefs(self):
-                    color_btn = self.glade.get_widget("%s_%s_color" % (graph, value))
-                    color_btn.set_color(gtk.gdk.Color(color))
+                    color_btn = self.main_builder.get_object("%s_%s_color" % (graph, value))
+                    col_gdk_color = eval(color).to_color()
```
`eval` does not seem good replacement code

doadin replied:
not sure...needs review TODO

cas-- commented on deluge/plugins/WebUi/deluge/plugins/webui/gtkui.py:
```diff
@@ -27,9 +26,10 @@
     def enable(self):
-        self.glade = gtk.glade.XML(get_resource("config.glade"))
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("config.ui"))
```
Again `self.glade =` not needed

doadin replied:
fixed

cas-- commented on deluge/scripts/create_plugin.py:
```diff
@@ -238,7 +238,7 @@ def get_resource(filename):
 class GtkUI(GtkPluginBase):
     def enable(self):
-        self.glade = gtk.glade.XML(get_resource("config.glade"))
+        self.glade = Gtk.glade.XML(get_resource("config.glade"))
```
Rename `config.glade` to `config.ui` and corresponding code later in script, the glade code also needs converting to gtkbuilder...

doadin replied:
fixed both

cas-- commented on deluge/ui/console/modes/input_popup.py:
```diff
@@ -107,11 +107,10 @@ def __init__(self, parent, message, name, child, checked=False, additional_forma
         self.name = name
         self.checked = checked
         self.msglen = len(self.chkd_inact) + 1
-        self.child = child
```
What are the changes to this file? They are not pygi relevant.

doadin replied:
fixed

cas-- replied:
I'm not sure what fixed means here? This file's code changes should be in a separate branch and commits with explanation of the whole changeset.

doadin replied:
by fixed i mean ive made changes to this file in this case just a revert of this line

cas-- commented on deluge/ui/gtkui/aboutdialog.py:
```diff
@@ -7,29 +7,28 @@
 from deluge.common import get_pixmap, get_version, open_url_in_browser
 from deluge.ui.client import client
 from deluge.ui.gtkui.common import get_deluge_icon
+from gi.repository import GdkPixbuf, Gtk
```
This import line should be above deluge imports, isort should fix that though...

doadin replied:
fixed

cas-- commented on deluge/ui/gtkui/addtorrentdialog.py:
```diff
@@ -288,7 +288,7 @@ def prepare_file_store(self, files):
                 )
             self.add_files(None, split_files)
-        self.listview_files.expand_row("0", False)
+        self.listview_files.expand_all()
```
Why is `expand_all` replacing `expand_row`?

doadin replied:
commented... TODO

cas-- commented on deluge/ui/gtkui/addtorrentdialog.py:
```diff
@@ -901,7 +901,7 @@ def walk_tree(row):
                 # We need to re-expand the view because it might contracted
                 # if we change the root iter
-                self.listview_files.expand_row("0", False)
+                self.listview_files.expand_all()
```
Why is `expand_all` replacing `expand_row`?

doadin replied:
commented... TODO

cas-- commented on deluge/ui/gtkui/common.py:
```diff
@@ -75,7 +75,7 @@ def build_menu_radio_list(value_list, callback, pref_value=None, suffix=None, sh
         if suffix:
             item_text += " " + suffix
-        menuitem = gtk.RadioMenuItem(group, item_text)
+        menuitem = Gtk.RadioMenuItem(item_text, group)
```
This looks like a mistake?

doadin replied:
without this  change and the change you commented on bellow the status bar makes menus that instead of values just say " < RadioMenuItem object @ (GtkRadioMenuItem @) > "

cas-- replied:
`Gtk.RadioMenuItem.new_with_label` perhaps?

doadin replied:
menuitem = Gtk.RadioMenuItem.new_with_label(group, item_text) =
TypeError: Must be sequence, not RadioMenuItem

so maybe but needs aditional changes?

cas-- replied:
Ah sorry it appears in Python GTK3 they have swapped them: http://learngtk.org/tutorials/python_gtk3_tutorial/html/radiomenuitem.html

cas-- commented on deluge/ui/gtkui/common.py:
```diff
@@ -84,7 +84,7 @@ def build_menu_radio_list(value_list, callback, pref_value=None, suffix=None, sh
 
     if show_notset:
-        menuitem = gtk.RadioMenuItem(group, notset_label)
+        menuitem = Gtk.RadioMenuItem(notset_label)
```
why has the group been removed?

doadin replied:
above comment

cas-- replied:
group should not have been removed, just swapped...

cas-- commented on deluge/ui/gtkui/common.py:
```diff
@@ -277,7 +277,7 @@ def listview_replace_treestore(listview):
         listview: a listview backed by a treestore
     """
-    # From http://faq.pygtk.org/index.py?req=show&file=faq13.043.htp
+    # From http://faq.pyGtk.org/index.py?req=show&file=faq13.043.htp
```
revert this

doadin replied:
fixed

cas-- commented on deluge/ui/gtkui/dialogs.py:
```diff
@@ -328,21 +328,21 @@ def __init__(self, header, text="", unit_text="", icon=None, default=0, parent=N
-        adjustment_spin = gtk.Adjustment(value=-1, lower=-1, upper=2097151, step_incr=1, page_incr=10)
-        self.spinbutton = gtk.SpinButton(adjustment_spin)
+        hbox = Gtk.HBox(spacing=5)
+        alignment_spacer = Gtk.Alignment.new(1, 0.5, 1, 1)
```
Why these values? It should be `0, 0, 0, 0` looking at the default for [gtk.Alignment](http://www.pygtk.org/pygtk2reference/class-gtkalignment.html#constructor-gtkalignment).

Also it appears [gtk-alignment-new](https://developer.gnome.org/gtk3/stable/GtkAlignment.html#gtk-alignment-new) is deprecated.

doadin replied:
partially fixed still in place but values changed... TODO

cas-- commented on deluge/ui/gtkui/dialogs.py:
```diff
@@ -328,21 +328,21 @@ def __init__(self, header, text="", unit_text="", icon=None, default=0, parent=N
+        hbox = Gtk.HBox(spacing=5)
+        alignment_spacer = Gtk.Alignment.new(1, 0.5, 1, 1)
+        hbox.pack_start(alignment_spacer, True, True, 0)
+        alignment_spin = Gtk.Alignment.new(1, 0.5, 1, 1)
```
Appears [gtk-alignment-new](https://developer.gnome.org/gtk3/stable/GtkAlignment.html#gtk-alignment-new) is deprecated.

doadin replied:
TODO

cas-- commented on deluge/ui/gtkui/dialogs.py:
```diff
@@ -328,21 +328,21 @@ def __init__(self, header, text="", unit_text="", icon=None, default=0, parent=N
+        alignment_spacer = Gtk.Alignment.new(1, 0.5, 1, 1)
+        hbox.pack_start(alignment_spacer, True, True, 0)
+        alignment_spin = Gtk.Alignment.new(1, 0.5, 1, 1)
+        adjustment_spin = Gtk.Adjustment.new(50.0, -1.0, 100000.0, 1.0, 10.0, 0.0)
```
Why have these values been changed?

doadin replied:
same as expand_all... TODO

cas-- commented on deluge/ui/gtkui/dialogs.py:
```diff
@@ -328,21 +328,21 @@ def __init__(self, header, text="", unit_text="", icon=None, default=0, parent=N
+        hbox.pack_start(alignment_spacer, True, True, 0)
+        alignment_spin = Gtk.Alignment.new(1, 0.5, 1, 1)
+        adjustment_spin = Gtk.Adjustment.new(50.0, -1.0, 100000.0, 1.0, 10.0, 0.0)
+        self.spinbutton = Gtk.SpinButton.new(adjustment_spin, 5.0, 0)
```
Why has this been changed, `5.0` is not the default, it should be `0`

doadin replied:
fixed

cas-- commented on deluge/ui/gtkui/files_tab.py:
```diff
@@ -810,7 +807,7 @@ def _on_drag_data_get_data(self, treeview, context, selection, target_id, etime)
     def _on_drag_data_received_data(self, treeview, context, x, y, selection, info, etime):
         try:
-            selected = cPickle.loads(selection.data)
+            selected = cPickle.loads(selection)
```
Are you sure this is correct?

doadin replied:
fixed

cas-- commented on deluge/ui/gtkui/files_tab.py:
```diff
@@ -810,7 +807,7 @@ def _on_drag_data_get_data(self, treeview, context, selection, target_id, etime)
-            selected = cPickle.loads(selection.data)
+            selected = cPickle.loads(selection)
         except cPickle.UnpicklingError:
             log.debug("Invalid selection data: %s", selection.data)
```
This would need changing too if the above change is made.

doadin committed:
Revert "GObject related things"

This reverts commit a0ee6aac92f9bb88a86f8d921e4c76394885829e.

doadin committed:
Address some issues commented by Cas

doadin committed:
add comment

cas-- commented on deluge/plugins/AutoAdd/deluge/plugins/autoadd/gtkui.py:
```diff
@@ -40,132 +39,134 @@ class OptionsDialog():
-        self.glade = gtk.glade.XML(get_resource("autoadd_options.glade"))
-        self.glade.signal_autoconnect({
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("autoadd_options.ui"))
```
self.glade not needed again

doadin replied:
ive done that so many times ....>.< sigh

cas-- commented on deluge/plugins/AutoAdd/deluge/plugins/autoadd/gtkui.py:
```diff
@@ -299,43 +299,43 @@ def on_cancel(self, event=None):
-        options['label'] = self.glade.get_widget('label').child.get_text().lower()
-        options['append_extension'] = self.glade.get_widget('append_extension').get_text()
+        #  AttributeError: 'ComboBox' object has no attribute 'child'
+        # options['label'] = self.main_builder.get_object('label').child.get_text().lower()
```
Have you tried replacing `child` for `get_child()`?

doadin replied:
no i have not... TODO

cas-- replied:
Please use FIXME instead of TODO: https://www.python.org/dev/peps/pep-0350/#mnemonics

doadin replied:
# options ['label'] = self.main_builder.get_object('label').get_child().get_text().lower() 

AttributeError: 'CellView' object has no attribute 'get_text'

cas-- commented on deluge/plugins/Extractor/deluge/plugins/extractor/gtkui.py:
```diff
@@ -26,9 +25,10 @@
     def enable(self):
-        self.glade = gtk.glade.XML(get_resource("extractor_prefs.glade"))
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("extractor_prefs.ui"))
```
self.glade again

cas-- commented on deluge/plugins/Extractor/deluge/plugins/extractor/gtkui.py:
```diff
@@ -37,36 +37,35 @@ def disable(self):
         component.get("Preferences").remove_page(_("Extractor"))
         component.get("PluginManager").deregister_hook("on_apply_prefs", self.on_apply_prefs)
         component.get("PluginManager").deregister_hook("on_show_prefs", self.on_show_prefs)
-        del self.glade
```
Probably should be `del self.main_builder`?

doadin replied:
fixed (changed)

cas-- commented on deluge/plugins/Blocklist/deluge/plugins/blocklist/gtkui.py:
```diff
@@ -141,11 +141,12 @@ def _on_status_item_clicked(self, widget, event):
         # Load the preferences page
-        self.glade = gtk.glade.XML(common.get_resource("blocklist_pref.glade"))
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("blocklist_pref.ui"))
```
self.glade again

cas-- commented on deluge/plugins/Label/deluge/plugins/label/gtkui/label_config.py:
```diff
@@ -29,9 +30,10 @@ def __init__(self, plugin):
         log.debug('Adding Label Preferences page')
-        self.glade = gtk.glade.XML(self.get_resource("label_pref.glade"))
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("label_pref.ui"))
```
self.glade again

cas-- commented on deluge/plugins/Label/deluge/plugins/label/gtkui/sidebar_menu.py:
```diff
@@ -112,18 +110,19 @@ def __init__(self):
-        self.glade = gtk.glade.XML(get_resource("label_options.glade"))
-        self.dialog = self.glade.get_widget("dlg_label_add")
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("label_options.ui"))
```
self.glade

cas-- commented on deluge/plugins/Label/deluge/plugins/label/gtkui/sidebar_menu.py:
```diff
@@ -152,19 +151,20 @@ def __init__(self):
-        self.glade = gtk.glade.XML(get_resource("label_options.glade"))
-        self.dialog = self.glade.get_widget("dlg_label_options")
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("label_options.ui"))
```
self.glade

cas-- commented on deluge/plugins/Notifications/deluge/plugins/notifications/gtkui.py:
```diff
@@ -231,11 +229,12 @@ def __init__(self, plugin_name):
-        self.glade.get_widget("smtp_port").set_value(25)
-        self.prefs = self.glade.get_widget("prefs_box")
+        self.main_builder = Gtk.Builder()
+        self.glade = self.main_builder.add_from_file(get_resource("config.ui"))
```
self.glade

cas-- commented on deluge/scripts/create_plugin.py:
```diff
@@ -238,7 +238,7 @@ def get_resource(filename):
 class GtkUI(GtkPluginBase):
     def enable(self):
-        self.glade = gtk.glade.XML(get_resource("config.glade"))
+        self.glade = Gtk.glade.XML(get_resource("config.ui"))
```
This should be builder code like the other plugins:

```
self.main_builder = Gtk.Builder()
self.main_builder.add_from_file(get_resource("config.ui"))
```

This likely will not be the only change needing done in this script.

doadin replied:
Yea i changed some of the other parts, but ill give it a look through for anything remaining.

cas-- commented on deluge/ui/gtkui/aboutdialog.py:
```diff
@@ -7,29 +7,28 @@
     def __init__(self):
         def url_hook(dialog, url):
             open_url_in_browser(url)
-        gtk.about_dialog_set_url_hook(url_hook)
```
Does removing this break clicking on the website links?

Also if you remove this line then also need to remove above `url_hook` function.

doadin replied:
from what i can tell no it doesnt. However i think we need a workaround of sorts because clicking the website link it pops up a windows saying no application to handle this or something like that. However the website items in the help menu open the browser just fine.

cas-- commented on deluge/ui/gtkui/files_tab.py:
```diff
@@ -367,20 +364,19 @@ def add_files(self, parent_iter, split_files):
     def update_files(self):
         with listview_replace_treestore(self.listview):
             self.prepare_file_store(self.files_list[self.torrent_id])
-        self.listview.expand_row("0", False)
```
Should be commented out with `# FIXME:` instead: https://www.python.org/dev/peps/pep-0350/#examples

cas-- commented on deluge/ui/gtkui/filtertreeview.py:
```diff
@@ -244,20 +241,24 @@ def get_pixmap(self, cat, value):
+        # pix.fill(0x0000000)
+        # return pix
+
+        # Should really give back a properly size pixbuf...
```
What does all this mean, is it fixed or not? Is the `icon_empty` code required or just a nice addition?

doadin replied:
i believe bendikro fixed this part and yes i think it is fixed so this commented part can be removed.

cas-- commented on win32/deluge-bbfreeze.py:
```diff
@@ -19,11 +18,11 @@
 # Include python modules not picked up automatically by bbfreeze
-includes = ("libtorrent", "cairo", "pangocairo", "atk", "pango", "twisted.internet.utils",
-            "gio", "gzip", "email.mime.multipart", "email.mime.text")
+includes = ("libtorrent", "cairo", "twisted.internet.utils",
```
You have removed `pango` but left in `cairo`?

doadin replied:
pango/pangocairo are in gi so they get imported by bbfreeze there. cairo is a seperate module and is still being used.

eg. piecesbar.py "import cairo" and "from gi.repository import Gdk, Gtk, Pango, PangoCairo"

cas-- replied:
You can use `from gi.repository import cairo`

doadin replied:
cool didnt know fixed.

cas-- commented on win32/deluge-bbfreeze.py:
```diff
@@ -3,12 +3,11 @@
 import icon
-from bbfreeze import Freezer
+from bbfreeze import Freezer, recipes  # NOQA
+from gi.repository import Gtk  # NOQA
```
Why are there `# NOQA` here?

doadin replied:
because other wise isort/flake fails. honeslty though im not sure why they are there but i just updated the import for the old import gtk so i just updated the import and left it alone.

cas-- replied:
Please do not use `#NOQA` it is failing because you have made a mistake. `flake8` is trying to help you write good code and `isort` puts the import lines in the correct place.

cas-- commented on win32/deluge-bbfreeze.py:
```diff
@@ -84,7 +83,6 @@ def ignored_files(adir, filenames):
 # copy gtk theme files
 theme_include_list = [
     [gtk_root, "share/icons/hicolor/index.theme"],
-    [gtk_root, "lib/gtk-2.0/2.10.0/engines"],
```
This surely means the entire `DelugeStart Theme` needs converting/removing/replacing?

doadin replied:
i believe so yes.

cas-- replied:
Ok well remove `theme_include_list` and the `for loop` below

cas-- commented on tox.ini:
```diff
@@ -183,4 +186,4 @@ deps =
     mkdir -p build/doccoverage
     sphinx-build -W -b coverage -d build/doctrees source build/doccoverage
-    py.test --doctest-glob='*.rst'
+    py.test --doctest-glob='*.rst'
```
No newline at end of file

cas-- commented on tox.ini:
```diff
@@ -37,7 +38,7 @@ commands = py.test deluge
 ##############
 # Unit tests
-##############
+############## #    DISPLAY = :99.0
```
Why is this comment here?

doadin replied:
bendikro did it in testing, the tox changes just need to be reverted now that his test fixes are merged in develop.

cas-- commented on docs/source/conf.py:
```diff
@@ -52,7 +52,7 @@ def __getattr__(cls, name):
                 'twisted.web', 'twisted.web.client', 'twisted.web.error',
                 'win32file', 'win32event', 'win32gui', 'win32api', 'win32con', '_winreg',
-                'pygtk', "gtk", "gobject", "gtk.gdk", "pango", "cairo", "pangocairo", "chardet"]
+                'pygtk', "gtk", "gobject", "Gtk.gdk", "pango", "cairo", "pangocairo", "chardet"]
```
This need checked/updated when building docs but for now add a # FIXME comment

cas-- commented on deluge/ui/gtkui/torrentview.py:
```diff
@@ -612,7 +645,12 @@ def on_button_press_event(self, widget, event):
+            # def pos(menu, icon):
+            #    return (Gtk.StatusIcon.position_menu(menu, icon))
+            # torrentmenu.popup_for_device(None, None, pos, self, event.button, event.time)
+            torrentmenu.popup(None, None, None, None, event.button, event.time)
```
Delete commented code?

cas-- commented on deluge/ui/gtkui/torrentview.py:
```diff
@@ -10,19 +10,22 @@
+from gi.repository import Gdk, GObject, Gtk
 
-pygtk.require('2.0')
+warnings.filterwarnings('error', category=UnicodeWarning)
```
We should not be filtering UnicodeWarning

cas-- commented on deluge/ui/gtkui/torrentdetails.py:
```diff
@@ -90,13 +89,13 @@ def __init__(self):
-            "Peers": _("_Peers"),
-            "Options": _("_Options"),
-            "Trackers": _("_Trackers")
+            "All": _("All"),
```
What is the code change here?

cas-- replied:
Why has the `_` been removed from these?

doadin replied:
honestly i just thought it looked better in the gui, why have _ in menu items? is there a conflict somewhere?

cas-- replied:
Those underscores are mnemonics so are meant to be there, see my comments further down code.

cas-- commented on deluge/ui/gtkui/systemtray.py:
```diff
@@ -103,14 +102,19 @@ def enable(self):
-                except:
-                    log.warning("Update PyGTK to 2.10 or greater for SystemTray..")
-                    return
+            self.tray = Gtk.StatusIcon.new_from_pixbuf(get_logo(32))
```
These changes are outside the scope of this branch, does the old code work?

doadin replied:
yea not sure why bendikro changed it

cas-- commented on deluge/ui/gtkui/path_combo_chooser.py:
```diff
@@ -1300,9 +1318,10 @@ def _on_entry_text_key_press_event(self, widget, event):
         Return True whenever we want no other event listeners to be called.
 
         """
+        print("on_entry_text_key_press_event Errors follow here when pressing ALT key while popup is visible")
```
This should be removed or be a comment

cas-- commented on deluge/ui/gtkui/mainwindow.py:
```diff
@@ -294,7 +295,7 @@ def on_vpaned_position_event(self, obj, param):
 
     def on_drag_data_received_event(self, widget, drag_context, x, y, selection_data, info, timestamp):
-        log.debug("Selection(s) dropped on main window %s", selection_data.data)
+        log.debug("Selection(s) dropped on main window %s", selection_data)  # fixed AttributeError: 'SelectionData' object has no attribute 'data'
```
I don't think that is the correct fix, it think it's likely to be: `selection_data.get_text()`

cas-- commented on deluge/ui/gtkui/mainwindow.py:
```diff
@@ -339,10 +340,17 @@ def is_on_active_workspace(self):
+        xid = self.window.get_window().get_xid()
+        win = Wnck.Window.get(xid)
+
+        active_wksp = screen.get_active_workspace()
```
Should this not be `active_wksp = win.get_screen().get_active_workspace()`? Would negate the need for line 349.

doadin replied:
not sure honestly bendikro wrote this part.

cas-- replied:
Ok so please leave a FIXME with my comment...

cas-- commented on deluge/ui/gtkui/mainwindow.py:
```diff
@@ -106,6 +105,7 @@ def patched_connect_signals(*a, **k):
         self.window = self.main_builder.get_object("main_window")
 
         self.window.set_icon(deluge.ui.gtkui.common.get_deluge_icon())
+        self.window.set_keep_above(False)
```
Why has this been added?

doadin replied:
in my testing the mainwindow kept going on top of the preferences window and maybe others...this could have been an issue with other things i was working on though... so ill try and test see what happens.

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -518,8 +539,9 @@ def add_column(self, header, render, col_types, hidden, position,
         column.set_visible(not hidden)
-        column.connect('button-press-event',
-                       self.on_treeview_header_right_clicked)
+        # column.connect('button-press-event',
```
What's the issue here? Add a FIXME if can't fix.

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -320,9 +333,17 @@ def on_menuitem_toggled(self, widget):
+        print "on_treeview_header_right_clicked"
         if event.button == 3:
-            self.menu.popup(None, None, None, event.button, event.get_time())
+            # self.menu.popup(None, None, None, event.button, event.get_time())
```
remove commented code

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -320,9 +333,17 @@ def on_menuitem_toggled(self, widget):
+            # self.menu.popup(None, None, pos, self, event.button, event.time)
+            # self.menu.popup(None, None, None, self, event.button, event.time)
+            self.menu.popup(None, None, None, None, event.button, event.get_time())
+            pass
```
remove pass

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -320,9 +333,17 @@ def on_menuitem_toggled(self, widget):
             self.columns[unicode(name)].column.set_visible(widget.get_active())
         return
 
+        # def callback(treeviewcolumn, user_param1, ...)'
```
remove

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -320,9 +333,17 @@ def on_menuitem_toggled(self, widget):
+        # clicked
+
     def on_treeview_header_right_clicked(self, column, event):
+        print "on_treeview_header_right_clicked"
```
remove

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -90,12 +96,13 @@ def __init__(self, title=None, cell_renderer=None, ** args):
 
         def on_button_pressed(self, widget, event):
-            self.emit('button-press-event', event)
+            # self.emit('button-press-event', event)
```
FIXME?

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -174,7 +181,12 @@ def create_model_filter(self):
-        self.model_filter = gtk.TreeModelSort(model_filter)
+        self.model_filter = Gtk.TreeModelSort(model_filter)
+
+        # print "self.model_filter:", type(self.model_filter)
```
remove?

cas-- commented on deluge/ui/gtkui/listview.py:
```diff
@@ -187,8 +199,9 @@ def set_model_sort(self):
-            self.model_filter.set_sort_column_id(self.default_sort_column_id, gtk.SORT_ASCENDING)
-        self.model_filter.set_default_sort_func(None)
+            self.model_filter.set_sort_column_id(self.default_sort_column_id, Gtk.SortType.ASCENDING)
+        # self.model_filter.set_default_sort_func(self.generic_sort_func, None)
```
FIXME?

cas-- commented on deluge/ui/gtkui/gtkui.py:
```diff
@@ -259,10 +263,10 @@ def nsapp_open_file(osxapp, filename):
 
         # Initialize gdk threading
-        gtk.gdk.threads_enter()
+        # Gdk.threads_enter()
```
why commented out?

cas-- replied:
delete the code

cas-- commented on deluge/ui/gtkui/gtkui.py:
```diff
@@ -259,10 +263,10 @@ def nsapp_open_file(osxapp, filename):
         reactor.run()
         self.shutdown()
-        gtk.gdk.threads_leave()
+        # Gdk.threads_leave()
```
why commented out?

cas-- replied:
delete the code

cas-- commented on deluge/ui/gtkui/gtkui.py:
```diff
@@ -270,8 +274,8 @@ def shutdown(self, *args, **kwargs):
-        while gtk.events_pending():
-            gtk.main_iteration(0)
+        # while Gdk.events_pending:
+        #    Gtk.main_iteration()
```
why commented out?

cas-- commented on deluge/ui/gtkui/gtkui.py:
```diff
@@ -212,7 +211,12 @@ def on_die(*args):
+        # gdk_threads_init has been deprecated since version 3.6
+        # and should not be used in newly-written code.
+        # All GDK and GTK+ calls should be made from the main thread
+        # Gdk.threads_init()
```
There are the two commits [64a94ec197ed19a760df3cc0aa8cfbc9b602f2d8] [1f3e930df31060804a119a34f6bb4dc37805ac92] that added gdk threading and I am thinking there is no need for the code anymore.

Delete the code but please create a good commit message including all this information.

doadin committed:
[GTKUI] Remove old threading code

threading in this way does not work and has been replaced by already
merged development code here:
https://github.com/doadin/deluge/commit/64a94ec197ed19a760df3cc0aa8cfbc9b602f2d8
and here:
https://github.com/doadin/deluge/commit/1f3e930df31060804a119a34f6bb4dc37805ac92

doadin committed:
Address some issues commented by Cas part 2

cas-- commented on deluge/ui/gtkui/torrentdetails.py:
```diff
@@ -256,9 +255,9 @@ def show_tab(self, tab_name, generate_menu=True):
+        menu = Gtk.Menu()
         # Create 'All' menuitem and a separator
-        menuitem = gtk.CheckMenuItem(self.translate_tabs["All"], True)
+        menuitem = Gtk.CheckMenuItem(self.translate_tabs["All"], True)
```
Replace with [Gtk.CheckMenuItem.new_with_mnemonic()](https://developer.gnome.org/gtk3/stable/GtkCheckMenuItem.html#gtk-check-menu-item-new-with-mnemonic)

cas-- commented on deluge/ui/gtkui/torrentdetails.py:
```diff
@@ -281,7 +280,7 @@ def generate_menu(self):
 
         for pos, name in menuitem_list:
-            menuitem = gtk.CheckMenuItem(self.translate_tabs[name], True)
+            menuitem = Gtk.CheckMenuItem(self.translate_tabs[name], True)
```
Replace with [Gtk.CheckMenuItem.new_with_mnemonic()](https://developer.gnome.org/gtk3/stable/GtkCheckMenuItem.html#gtk-check-menu-item-new-with-mnemonic)

doadin replied:
breaks the label preference tab being added doesnt give a  error. added as FIXME.

cas-- commented on deluge/plugins/Label/deluge/plugins/label/gtkui/sidebar_menu.py:
```diff
@@ -39,12 +37,12 @@ def __init__(self):
-        self._add_item("options", _("Label _Options"), gtk.STOCK_PREFERENCES)
-        self._add_item("remove", _("_Remove Label"), gtk.STOCK_REMOVE)
-        self._add_item("add", _("_Add Label"), gtk.STOCK_ADD)
+        self._add_item("options", _("Label Options"), Gtk.STOCK_PREFERENCES)
```
mnemonics, please revert and fix code as discussed

cas-- commented on deluge/plugins/Label/deluge/plugins/label/gtkui/sidebar_menu.py:
```diff
@@ -39,12 +37,12 @@ def __init__(self):
-        self._add_item("remove", _("_Remove Label"), gtk.STOCK_REMOVE)
-        self._add_item("add", _("_Add Label"), gtk.STOCK_ADD)
+        self._add_item("options", _("Label Options"), Gtk.STOCK_PREFERENCES)
+        self._add_item("remove", _("Remove Label"), Gtk.STOCK_REMOVE)
```
mnemonics, please revert and fix code as discussed

cas-- commented on deluge/common.py:
```diff
@@ -957,17 +963,16 @@ def setup_translations(setup_gettext=True, setup_pygtk=False):
-            import gtk.glade
-            gtk.glade.bindtextdomain(domain, translations_path)
-            gtk.glade.textdomain(domain)
+            # Use glade for plugins that still use it
```
Does translation still work under Linux? Including in the plugins

doadin replied:
havnt confirmed linux

cas-- commented on deluge/plugins/Execute/deluge/plugins/execute/data/execute_prefs.ui:
```diff
@@ -124,45 +145,46 @@
+                    <property name="can_focus">False</property>
                     <property name="events">GDK_POINTER_MOTION_MASK | GDK_POINTER_MOTION_HINT_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK</property>
-                    <property name="orientation">vertical</property>
+					<property name="orientation">vertical</property>
```
Too much indent here

cas-- commented on win32/deluge-bbfreeze.py:
```diff
@@ -19,11 +18,11 @@
 
 print "Python Path: %s" % python_path
-gtk_root = os.path.join(gtk.__path__[0], "..\\runtime\\")
+gtk_root = "C:\\Python27\\Lib\\site-packages\\gnome"
```
This is not correct

cas-- replied:
We need the Gtk root folder to get locale files and I'm not sure this is it. Please add a FIXME comment.

doadin replied:
This is correct but i do agree that re-writing this to not be hardcoded would be better.

cas-- commented on win32/deluge-bbfreeze.py:
```diff
@@ -3,12 +3,11 @@
-import gtk
-
-import bbfreeze.recipes
+import bbfreeze
```
Revert this change

cas-- commented on win32/deluge-bbfreeze.py:
```diff
@@ -3,12 +3,11 @@
 import deluge.common
 import icon
-from bbfreeze import Freezer
+from bbfreeze import Freezer, recipes  # NOQA
```
Revert this change

cas-- commented on deluge/ui/gtkui/mainwindow.py:
```diff
@@ -339,10 +340,17 @@ def is_on_active_workspace(self):
+
+        # TODO: Linux specific code required? Above code fails on Ubuntu 14.04...
+        # Necessary to import this for get_xid() to work, at least when testing on Ubuntu 14.04
+        from gi.repository import Gtk, GdkX11  # NOQA
```
No need to import `Gtk` again here.

cas-- commented on docs/source/conf.py:
```diff
@@ -52,7 +52,7 @@ def __getattr__(cls, name):
                 'twisted.web', 'twisted.web.client', 'twisted.web.error',
                 'win32file', 'win32event', 'win32gui', 'win32api', 'win32con', '_winreg',
-                'pygtk', "gtk", "gobject", "gtk.gdk", "pango", "cairo", "pangocairo", "chardet"]
+                'pygtk', "gtk", "gobject", "Gtk.gdk", "pango", "cairo", "pangocairo", "chardet"]  # FIXME
```
You have to leave more information than just FIXME, also don't put comments at the end of the line... https://www.python.org/dev/peps/pep-0350/#examples

doadin committed:
Another Cas Comment wave

doadin committed:
Fix translating

fix from going from one translation to another then back again

doadin committed:
remove print dd back commented code

doadin committed:
Change TODO to FIXME: mnemonics

doadin committed:
Autoadd add fixme and update comment

doadin committed:
GTKUI listview fix default sort

doadin committed:
GTKUI listview remove comment from button press event

doadin committed:
GTKUI mainwindow remove testing code

doadin committed:
GTKUI mainwindows add FIXME comment from cas

doadin committed:
GTKUI mainwindow add fixme for on_drag_data_received_event

doadin committed:
GTKUI mainwindow add fix from cas

doadin committed:
Plugin Label revert change mnemonics

doadin committed:
GTKUI systemtray revert change

cas  "outside the scope of this branch"

doadin committed:
GTKUI torrentdetails revert change mnemonics

doadin committed:
tox remove comment

doadin committed:
Fix line end

doadin committed:
cleanup bbfreeze script

remove missing gtk3 theme parts

doadin committed:
GTKUI gtkui add fixme comment

doadin committed:
GTKUI filtertreeview remove commented code

doadin committed:
update peicesbar cairo import and bbfreeze include cairo

doadin committed:
more updates to create_plugin script

doadin committed:
GTKUI common revert removing group

doadin committed:
plugin stats fixme with comment

doadin committed:
plugin stats fixme with comments

doadin committed:
rename listview_files to treeview_files

cas "To make it clearer what is being references in code, should rename
listview_files to treeview_files and obviously change all references in
deluge/ui/gtkui/addtorrentdialog.py"

doadin committed:
plugin notifications fixme with comment

doadin committed:
add removed code comment with fixme

"File "C:\Python27\lib\site-packages\gi\overrides\GObject.py", line 545,
in _un
supported_data_method
raise RuntimeError('Data access methods are unsupported. '
RuntimeError: Data access methods are unsupported. Use normal Python
attributes instead"

same for all set_data lines

doadin committed:
fix error in notifications gtkui from convert script

cas-- committed:
get_introspection_module does not seen to be available on ubuntu 12.04

doadin committed:
add missing expand_row still fixme

cas-- committed:
Fixes for wnck, events_pending, add FIXMEs and some corrections

cas-- committed:
Fix model_filer and add more FIXME comments

cas-- committed:
Flake8 and isort code pass

doadin committed:
Fix GTKUI dialogs.py pack_start error

doadin committed:
Fix one fixme? add another...

doadin committed:
GTKUI preference fix type error

Fixes plugins not enableing on systems with older versions of gtk

doadin committed:
Plugin blocklist fix progressbar

doadin committed:
plugin blocklist fix spacing in ui file

doadin committed:
fix label plugin menu because of error in filtertreeview

fixes main ui right-click menu doesnt show in older versions of gtk due
to error in conversion

doadin committed:
plugin blocklist fix disable

doadin committed:
plugin autoadd fix error dialog not showing

doadin committed:
plugin autoadd remove unused handel

doadin committed:
Plugins fix autoadd and blocklist

* remove unused handel from autoadd
* fix progressbar text not showing

doadin committed:
Possible files tab fix

doadin committed:
Revert "Possible files tab fix"

This reverts commit df3cfd8bf32a8b8a76ed6267f0fc5b3639ce8520.

doadin committed:
GTKUI preferences_dialog convert GtkTable to GtkGrid

* GtkTable is deprecated since 3.0
* need to adjust spacing in alot of text

doadin committed:
GTKUI fix most preferences text alignment issues

* on gtk < 3.10 text was miss aligned this fixes most visual issues on
older versions and newer so they look the same and proper.
* Only issue left is the Preferences > Network > Encryption > Level
lebel/dropdown menu are too far right only in older versions of gtk just
like the other items fixed in this commit. However havnt found the
change thats needed.

doadin committed:
plugins fix blocklist handelers

doadin committed:
plugins fix notifications handelers and adding preference page

doadin committed:
fix systemtray error

doadin committed:
Revert "fix systemtray error"

This reverts commit 1c73969f68dc0edc4ef4ad1551e7a2dafe2012fa.

doadin committed:
plugins fix text alignment

doadin committed:
plugin notifications fix missed alignment
